app/build.gradle.kts

```
@@ -44,18 +44,21 @@ android {
             isMinifyEnabled = true
             proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
         }
-        val benchmark by creating {
-            initWith(release)
-            signingConfig = signingConfigs.getByName("debug")
-            matchingFallbacks.add("release")
-            proguardFiles("benchmark-rules.pro")
-        }
         val staging by creating {
             initWith(debug)
             signingConfig = signingConfigs.getByName("debug")
             matchingFallbacks.add("debug")
             applicationIdSuffix = ".staging"
         }
+        val benchmark by creating {
+            initWith(staging) // Usually should be `initWith(release)`. Connecting to demo backend.
+            matchingFallbacks.add("debug") // Making some settings below to align closer to release.
+            signingConfig = signingConfigs.getByName("debug")
+            proguardFiles("benchmark-rules.pro") // Only use benchmark proguard rules
```

We want to use both app's rules + benchmark rules (that has just -dontobfuscate). That's probably why isMinifyEnabled = true doesn't work

```suggestion
            proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro", "benchmark-rules.pro")
```

---

Minification isn't the problem, but as soon as `minifyEnabled true` the connection to the data breaks.